packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx

```diff
@@ -0,0 +1,78 @@
+import {Component, ContextType} from 'react';
+import {
+  buildStandaloneSearchBox,
+  StandaloneSearchBox as HeadlessStandaloneSearchBox,
+  StandaloneSearchBoxState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class StandaloneSearchBox extends Component<
+  {},
+  StandaloneSearchBoxState
+> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessStandaloneSearchBox;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildStandaloneSearchBox(this.context.engine!, {
+      options: {redirectionUrl: 'https://mywebsite.com/search'},
+    });
+    this.updateState();
+
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  componentDidUpdate() {
+    if (!this.state?.redirectTo) {
+      return;
+    }
+    window.location.href = `${this.state.redirectTo}/${encodeURIComponent(
+      this.state.value
```

Comment:
I would remove the encoded part and just redirect to the exact redirection url in order not to mislead the user